src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FC } from "react";
+import { Field, Form } from "react-final-form";
+import { TODO } from "../../types";
+import { useAddTodoMutation } from "../../redux/apiSlice";
+import { StyledForm } from "./TodoFormStyles";
+import { StyledButton } from "../JustButton/JustButtonStyles";
+import { Send } from "tabler-icons-react";
+
+export const TodoForm: FC = () => {
+  const [addTodo] = useAddTodoMutation();
+  const onSubmit = (values: Pick<TODO, "text">) => {
+    addTodo(values);
+  };
+  const validator = (value: string) => (value ? null : "Required field");
+
+  return (
+    <Form
+      onSubmit={onSubmit}
+      initialValues={{ text: "" }}
+      render={(props) => (
+        <StyledForm
+          onSubmit={(event) => {
```

Comment:
To vypadá, že formulář resetuješ dřív než se ti dokončí submit. To znamená, že kdyby jsi měl chybu tak ho vyresetuješ musíš psát znovu. Není to dobrý přístup, měl by jsi to vyresetovat až na úspěšný submit.